pymongo/auth.py

```diff
@@ -458,6 +497,218 @@ def _authenticate_mongo_cr(credentials, sock_info):
     sock_info.command(source, query)
 
 
+# MONGODB-OIDC private variables.
```

Comment:
What do you think about turning the changes in this file into static methods in a private class like _OIDCContext, and then remove all the leading underscores from the methods and variable names.